### PR TITLE
feat: add setting to render images in comments

### PR DESCRIPTION
**Pull request by sharunkumar**
Resolves https://github.com/aeharding/voyager/issues/108

Disabled by default: https://github.com/aeharding/voyager/issues/108#issuecomment-1615435828

Also in this PR:
- Render compact post settings only when relevant setting is selected

### CODE REVIEW

File: src/features/settings/general/comments/RenderCommentImages.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export default function RenderCommentImages() {
+  const dispatch = useAppDispatch();
+  const { renderCommentImages } = useAppSelector(
+    // this needs a better naming
+    (state) => state.settings.general.comments,
+  );
+
+  return (
+    <InsetIonItem>
+      <IonLabel>Render Comment Images</IonLabel>
```
Comment (aeharding) [inline src/features/settings/general/comments/RenderCommentImages.tsx:15]: Lets rename everything from "render" comment images to "show" comment images

File: src/features/settings/appearance/CompactSettings.tsx [hunk 2]
```diff
@@ -18,6 +18,12 @@ export default function CompactSettings() {
   const { thumbnailsPosition, showVotingButtons, thumbnailSize } =
     useAppSelector((state) => state.settings.appearance.compact);
 
+  const postsAppearanceType = useAppSelector(
+    (state) => state.settings.appearance.posts.type,
+  );
+
+  if (!(postsAppearanceType === "compact")) return <></>;
```
Comment (aeharding) [inline src/features/settings/appearance/CompactSettings.tsx:25]: Lets still render if compact mode - its what apollo does :)

File: src/features/comment/CommentMarkdown.tsx [hunk 3]
```diff
@@ -6,19 +7,27 @@ interface CommentMarkdownProps {
 }
 
 export default function CommentMarkdown({ children }: CommentMarkdownProps) {
+  const { renderCommentImages } = useAppSelector(
+    (state) => state.settings.general.comments,
+  );
+
   return (
     <Markdown
```
Comment (aeharding) [inline src/features/comment/CommentMarkdown.tsx:15]: Can we make this Markdown a StyledMarkdown and add css to target `img` inside with a max-height? Perhaps 200px will do.

**Comment by aeharding** [comment]
One other thing - we should probably edit `CommentLinks` and filter out `type = 'image'` if you have show comment images turned on.

File: src/features/settings/appearance/CompactSettings.tsx (hunk 2 above)
Comment (sharunkumar) [reply src/features/settings/appearance/CompactSettings.tsx:25]: You mean, when compact mode is off?

Comment (aeharding) [reply src/features/settings/appearance/CompactSettings.tsx]: Yeah, I would like to keep the previous behavior where the compact settings show all times.

There will probably be a feature in the future where you can change compact to large post mode in the feed.

**Pull request merged**